### PR TITLE
[stable/airflow] update to airflow 1.10.10

### PR DESCRIPTION
**Pull request by thesuperzapper**
This is a very significant update for the chart, with changes across every template.

Broadly, it implements support for Airflow 1.10.10, and the new `apache/airflow` Docker images, this means we have removed support for `puckel/docker-airflow` images. 

Additionally, I have cleaned up the chart's syntax to aid readability, improved the docs, added new functionality, and squashed many bugs.

There are many breaking changes included, (given this is a major version bump), and these are documented under the `7.0.0` upgrading docs in `README.md`.

Naturally given the scale of the changes, testing is hard, but I have successfully deployed this in my GKE cluster, with multiple custom value arrangements. I believe this PR is in a state in which it can be merged, and we can hotfix any issues that crop up.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by thesuperzapper** [comment]
Hi @gsemet I know this is a bit of a behemoth, but can you review it?

File: stable/airflow/examples/minikube-values.yaml [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
 
-  variables: '{ "environment": "dev" }'
-  pools: '{ "example": { "description": "This is an example of a pool", "slots": 2 } }'
+web:
+  service:
+    type: NodePort
+
+scheduler:
+  variables: |
+    { "environment": "dev" }
```
Comment (gsemet) [inline stable/airflow/examples/minikube-values.yaml:18]: Can you add a comment this actually a string ?

File: stable/airflow/examples/minikube-values.yaml [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
+  pools: |
```
Comment (gsemet) [inline stable/airflow/examples/minikube-values.yaml:21]: And here as well

**Comment by gsemet** [comment]
Ok this looks pretty huge ! I can see there is so much work on it. I may take the day to fully review it, but it looks awesome. Docs has been enhanced a lot and the upgrade section is accurate so user can update pretty easily... so thanks a lot ! And we finally ends up using the official docker image!

File: stable/airflow/values.yaml [hunk 3]
```diff
@@ -1,786 +1,1274 @@
...
-##      name: "{{ .Release.Name }}-test"
-##    spec:
-##      securityPolicy:
-##        name: "gcp-cloud-armor-policy-test"
+  ## alerting rules for Prometheus
+  ##
+  ## NOTE:
+  ## - documentation: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
+  ##
+  groups: []
```
Comment (gsemet) [inline stable/airflow/values.yaml]: you need to add a new line for the linter

File: stable/airflow/README.md [hunk 4]
```diff
@@ -1,208 +1,209 @@
...
-`AIRFLOW__<section>__<key>`.
-
-See the
-[Airflow documentation for more information](http://airflow.readthedocs.io/en/latest/configuration.html?highlight=__CORE__#setting-configuration-options)
-
-This helm chart allows you to add these additional settings with the value key `airflow.config`.
-You can also add generic environment variables such as proxy or private pypi:
+We expose the `scheduler.variables` value to allow specifying [Airflow Variables](https://airflow.apache.org/docs/stable/concepts.html#variables) at deployment time, variables will be automatically imported by the Airflow scheduler when it starts up.
 
+For example, to specify an `environment` variable:
```
Comment (gsemet) [inline stable/airflow/README.md:72]: can you add a remark the `variables` value is a string and not a structure.

maybe use the | syntax:
```yaml
airflow:
  variables: |
    environment: "dev"

**Comment by gsemet** [comment]
lgtm, only a few remark on it. I think we can merge it and see how user react to this new version that break the existing configuration but I think it worth it. Do you want to be added in the OWNER or any Airflow developer for future contributions?

**Comment by thesuperzapper** [comment]
@gsemet yeah, I would love to be an OWNER, I have added that and the other changes if you want to go ahead and merge.

**Comment by gsemet** [comment]
Let's try and see how it is received. Thanks a lot for this big PR, using the official docker image is a major milestone :)

**Comment by marcosmarxm** [comment]
I agree that changing to the official image is necessary. However at this moment it seems that it is still quite young and will have several changes. In my opinion, maintaining a more stable image (puckel/airflow) at this moment would be more consistent until the official consolidates.

**Comment by gsemet** [comment]
What we can do it propose a new chart stable/airflow-legacy (or airflow-puckel) with code from 6.x.
I think we need to move on, airflow will propose its own chart based On another image (on Astronomers image I guess) soon, and wanted my help and the help of the community to move towards uniformisation of the images. So if we use the official image while still providing most of our feature a big chunk of the work will already be done and this will help the upstream project to maturate its docker image !

**Comment by kkourtchikov** [comment]
Thanks so much for taking on the Airflow upgrade effort!

When testing v7.0.1 of the chart, I noticed [this block](https://github.com/helm/charts/blob/5195adc2198ec6d184b73e84e21a9150c97948f1/stable/airflow/templates/deployments-scheduler.yaml#L186-L189) seems to break the function of using custom Airflow Docker images with DAGs included in them. If it is in fact the intention to mount an [emptyDir](https://github.com/helm/charts/blob/5195adc2198ec6d184b73e84e21a9150c97948f1/stable/airflow/templates/deployments-scheduler.yaml#L268) when persistence is disabled, as opposed to not mounting dags-data at all, what would you suggest be the best way to deploy Airflow via a docker image with the DAGs in it? I have found this docker image approach to be pretty common when using the KubernetesExecutor. Many thanks in advance!

**Comment by thesuperzapper** [comment]
@kkourtchikov I agree that change was not intended, I will incorporate a fix into the next version.

**Pull request merged**